File: static/src/fun_with_crystal/loop.rb
size = 10 ** 5
a = (1..size).map{ [rand(1000), rand(1000)] }

def dist(u,v)
  (u[0] - v[0]).abs + (u[1] - v[1]).abs
end

a.each_with_index {|u,i|
  print "\r#{i}" if i % 100 == 0
  a[i + 1..-1].each_with_index {|v,j|
    dist(u,v) < 3
  }
}

File: static/src/fun_with_crystal/loop.js
const size = 100000
const a = new Array(size).fill().map(() =>
  [Math.floor(Math.random()*1000), Math.floor(Math.random()*1000)])

const dist = (u,v) =>
  Math.abs(u[0] - v[0]) + Math.abs(u[1] - v[1])

a.forEach((u,i) => {
  i % 100 == 0 ? console.log(`\r${i}`) : null;
  a.slice(i + 1, size).forEach((v,i) => {
    dist(u,v) < 3
  })
})